### PR TITLE
Clean up shared styles

### PR DESCRIPTION
**Pull request by julieg18**
* removes `.storybook/styles.scss`
* renames `shared/style.scss` to `shared/styles.scss`
* removed unneeded import of `shared/styles.scss`
* adds `mixins.scss` for holding reusable mixins

Followup to #3696

### CODE REVIEW

File: webview/src/experiments/components/table/styles.module.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Variables
 
 @import '../../../shared/variables.scss';
-@import '../../../shared/style.scss';
+@import '../../../shared/styles.scss';
```
Comment (sroy3) [inline webview/src/experiments/components/table/styles.module.scss:4]: I don't think this line is necessary as it is already imported in `webview/src/experiments/index.tsx`. There doesn't seem to be any variables or classes to extend in there.

File: webview/src/shared/components/sectionContainer/styles.module.scss [hunk 2]
```diff
@@ -1,5 +1,5 @@
 @import '../../variables.scss';
-@import '../../style.scss';
+@import '../../styles.scss';
```
Comment (sroy3) [inline webview/src/shared/components/sectionContainer/styles.module.scss:2]: Same here, I don't think we need to import this.

Comment (julieg18) [reply webview/src/shared/components/sectionContainer/styles.module.scss:2]: Good catch!

File: webview/src/experiments/components/table/styles.module.scss (hunk 1 above)
Comment (julieg18) [reply webview/src/experiments/components/table/styles.module.scss:4]: Looks like we are extending an `a` element, which should be done in a mixin... I'll go ahead and create a `shared/mixins.scss` file similar to our `variables.scss` file.

**Pull request merged**